locations/paas/src/main/java/brooklyn/location/PaasProvisioningLocation.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package brooklyn.location;
+
+public interface PaasProvisioningLocation<T extends PaasLocation> extends ProvisioningLocation<T> {
```

Comment:
@andreaturli I think I did not understand your question. I did not want to use a ssh in `PaaSLocation`. 

We could define a interface to indicate that a location allows to instance and contain a PaaS Container Client (`PaaSClient`), for example [cf-client-lib](https://github.com/cloudfoundry/cf-java-client/tree/master/cloudfoundry-client-lib). This interface may be called PaaSClientProvisioningLocation (or similar, probably this name it is not the better) and it should provide some methods like:
```
PaaSClient setup(Map configParameters);
```
which allows to instance and setup the `PaaSClient` such as cf-client-lib.
Then, `CloudFoundryPaaSLocation` could extends `PaaSLocation` and implements `PaaSClientProvisioningLocation`.